aws/resource_aws_security_group.go

```diff
@@ -342,11 +332,12 @@ func resourceAwsSecurityGroupCreate(d *schema.ResourceData, meta interface{}) er
 func resourceAwsSecurityGroupRead(d *schema.ResourceData, meta interface{}) error {
 	conn := meta.(*AWSClient).ec2conn
 
-	sgRaw, _, err := SGStateRefreshFunc(conn, d.Id())()
+	sgRaw, err := waitForSgToExist(conn, d.Id(), d.Timeout(schema.TimeoutRead))
```

Comment:
I think in both the read/update cases we will want to check `d.IsNewResource()` before using the `waitForSgToExist` logic -- otherwise people will have to wait for the entire read timeout when it is removed outside Terraform.

Reply:
Excellent, will fix.